consensus/src/consensus.rs

```diff
@@ -503,6 +524,48 @@ impl<R: ConsensusRpc> ConsensusStateManager<R> {
         Ok(())
     }
 
+    fn bootstrap_offline(&mut self, checkpoint: &[u8], bootstrap: &mut Bootstrap) -> Result<()> {
+        let is_valid = self.is_valid_checkpoint(bootstrap.header.slot.into());
+
+        if !is_valid {
+            if self.config.strict_checkpoint_age {
+                return Err(ConsensusError::CheckpointTooOld.into());
+            } else {
+                warn!(target: "helios::consensus", "checkpoint too old, consider using a more recent block");
+            }
+        }
+
+        let committee_valid = is_current_committee_proof_valid(
```

Comment:
This code is from the original `helios` repo,  me and Zeev agreed we would not refactor code unless it is necessary to the logics we need to implement. 
If we do this, this would have no end, so we leave it that way.